src/client/common/client_common.cpp

```diff
@@ -76,28 +109,59 @@ std::string mp::cmd::update_notice(const mp::UpdateInfo& update_info)
                                      update_info.description(), update_info.url()));
 }
 
-std::shared_ptr<grpc::Channel> mp::client::make_channel(const std::string& server_address,
-                                                        mp::RpcConnectionType conn_type,
-                                                        mp::CertProvider& cert_provider)
+std::shared_ptr<grpc::Channel> mp::client::make_secure_channel(const std::string& server_address,
+                                                               mp::CertProvider* cert_provider)
 {
-    std::shared_ptr<grpc::ChannelCredentials> creds;
-    if (conn_type == mp::RpcConnectionType::ssl)
-    {
-        auto opts = grpc::SslCredentialsOptions();
-        opts.server_certificate_request = GRPC_SSL_REQUEST_SERVER_CERTIFICATE_BUT_DONT_VERIFY;
-        opts.pem_cert_chain = cert_provider.PEM_certificate();
-        opts.pem_private_key = cert_provider.PEM_signing_key();
-        creds = grpc::SslCredentials(opts);
-    }
-    else if (conn_type == mp::RpcConnectionType::insecure)
-    {
-        creds = grpc::InsecureChannelCredentials();
-    }
-    else
+    if (!cert_provider)
```

Comment:
This is only here for testing purposes, right? As we hold on to the `grpc::Channel` in the real clients?

Do we ever expect a client to come here more than once?

Reply:
Well, kind of testing purposes and also to support the whole transition from 2 separate client certificates to a single shared client certificate.

> Do we ever expect a client to come here more than once?

Not sure what you mean by this question.

Reply:
I mean that in real clients `cert_provider` will never be a valid pointer?

Reply:
I suppose one thing to do instead of always passing in `nullptr` from the client config is to first detect if the common client cert dir already exists, and if so, just use that in the passed in `cert_provider`.  It will make the deprecation of this code a little less painful and also will not go into this logic at all.

Reply:
I did clean up the code and it no longer just passes `nullptr` when constructing `ClientConfig` in `main_impl()`.  This particular logic is still here for the "migration" code, but it's now a bit simpler.